cmplibrary/src/test/java/com/sourcepoint/gdpr_cmplibrary/GDPRUserConsentTest.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    @Test
+    public void testUserConsentUUID() {
+
+        assertNotNull(userConsent.uuid);
+        assertEquals(consentUUID, userConsent.uuid);
+    }
+
+    @Test
+    public void testUserConsentConsentString() throws JSONException {
+
+        assertNotNull(userConsent.consentString);
+        assertEquals(userConsent.consentString, jsonConsents.getString("euconsent"));
+    }
+
+    @Test
+    public void testUserConsentAcceptedCategories() throws JSONException {
+
+        ArrayList<String> acceptedCategories = userConsent.json2StrArr(jsonConsents.getJSONArray("acceptedCategories"));
+
+        assertNotNull(userConsent.acceptedCategories);
+        assertEquals(acceptedCategories, userConsent.acceptedCategories);
+    }
+
+    @Test
+    public void testUserConsentAcceptedVendors() throws JSONException {
+        ArrayList<String> acceptedVendors = userConsent.json2StrArr(jsonConsents.getJSONArray("acceptedVendors"));
+
+        assertNotNull(userConsent.acceptedVendors);
+        assertEquals(acceptedVendors , userConsent.acceptedVendors);
+    }
+
+    @Test
+    public void testUserConsentLegInCategories() throws JSONException {
+        ArrayList<String> legIntCategories = userConsent.json2StrArr(jsonConsents.getJSONArray("legIntCategories"));
+
+        assertNotNull(userConsent.legIntCategories);
+        assertEquals(legIntCategories, userConsent.legIntCategories);
+    }
+
+    @Test
+    public void testUserConsentSpecialFeatures() throws JSONException {
+
+        ArrayList<String> specialFeatures = userConsent.json2StrArr(jsonConsents.getJSONArray("specialFeatures"));
+
+        assertNotNull(userConsent.specialFeatures);
+        assertEquals(specialFeatures, userConsent.specialFeatures);
+
+    }
+
+    @Test
+    public void testUserConsentTCData() throws JSONException, ConsentLibException {
+        HashMap tcData = CustomJsonParser.getHashMap(jsonConsents.getJSONObject("TCData"));
+
+        assertNotNull(userConsent.TCData);
+        assertEquals(tcData, userConsent.TCData);
+    }
+
+    @Test
+    public void testUserConsentVendorGrants() throws JSONException, ConsentLibException {
+        jsonConsents.put("uuid",consentUUID);
+
+        GDPRUserConsent.VendorGrants vendorGrants = new GDPRUserConsent(jsonConsents).vendorGrants;
+
+        assertNotNull( userConsent.vendorGrants);
+        assertEquals(vendorGrants.toString(), userConsent.vendorGrants.toString());
+    }
+
+    @Test
+    public void toJsonObjectTest() throws JSONException, ConsentLibException {
+
+        assertNotNull(userConsent.toJsonObject());
+        assertEquals(userConsent.toJsonObject().getClass() , JSONObject.class);
+    }
+
+    @Test
+    public void testUserConsentConstructorWithoutUUID() throws ConsentLibException, JSONException {
+        jsonConsents.put("uuid",consentUUID);
+
+        GDPRUserConsent userConsentWithoutUUID = new GDPRUserConsent(jsonConsents);
```

Comment:
this name is misleading. This userConsents has a uuid, but the constructor method used doesn't

Reply:
@leodoin This name indicates we are  testing GDPRUserConset class constructor without UUID.